lib/ddtrace/contrib/excon/middleware.rb

```diff
@@ -42,9 +42,8 @@ def response_call(datum)
         end
 
         def error_call(datum)
-          @stack.error_call(datum).tap do |d|
-            handle_response(d)
-          end
+          handle_response(datum)
+          @stack.error_call(datum)
```

Comment:
I see, we have to `handle_response` before calling the stack for `error_call`?

Reply:
Yes - see commit message for a more detailed explanation. There's also evidence in the Excon::Middleware::Base class indicating that this is the order it should be:

https://github.com/excon/excon/blob/master/lib/excon/middlewares/base.rb#L11